entity-event-generator-lambda/src/main/scala/uk/gov/nationalarchives/Lambda.scala

```diff
@@ -98,37 +81,54 @@ class Lambda extends RequestHandler[ScheduledEvent, Unit] {
           }
         } else IO(None)
 
-      lastEventActionBeforeEventTriggered = entityLastEventActionDate.map(_.isBefore(eventTriggeredDatetime))
-
-      _ <-
-        if (lastEventActionBeforeEventTriggered.getOrElse(false)) {
-          val compactEntities: List[CompactEntity] = convertToCompactEntities(recentlyUpdatedEntities.toList)
-
-          for {
-            _ <- dASnsDBClient.publish[CompactEntity](config.snsArn)(compactEntities)
-            updateDateAttributeValue = AttributeValue.builder().s(entityLastEventActionDate.get.toString).build()
-            updateDateRequest = DADynamoDbRequest(
-              config.lastEventActionTableName,
-              dateItemPrimaryKeyAndValue,
-              Map(datetimeField -> Some(updateDateAttributeValue))
-            )
-            statusCode <- dADynamoDBClient.updateAttributeValues(updateDateRequest)
-          } yield statusCode
-        } else IO(0)
+      _ <- IO.whenA(entityLastEventActionDate.exists(_.isBefore(eventTriggeredDatetime))) {
+        for {
+          _ <- dASnsDBClient.publish[CompactEntity](config.snsArn)(convertToCompactEntities(recentlyUpdatedEntities.toList))
+          updateDateAttributeValue = AttributeValue.builder().s(entityLastEventActionDate.get.toString).build()
+          updateDateRequest = DADynamoDbRequest(
+            config.lastEventActionTableName,
+            dateItemPrimaryKeyAndValue,
+            Map(datetimeField -> Some(updateDateAttributeValue))
+          )
+          _ <- dADynamoDBClient.updateAttributeValues(updateDateRequest)
```

Comment:
Do you think it's worth logging the `statusCode`?

Reply:
Yeah, makes sense, I've put a line in.